### ISSUE TITLE
Circular dependency between wickedd and openvswitch

### ISSUE THREAD

**Issue by jwgn** [sap-oc/crowbar-core#24]
When rebooting network nodes, they come up without network connection.

The problem: Start of wickedd depends on a running openvswitch. openvswitch needs bond0 to start. But bond0 is being created by wickedd. No bond0 = no openvswitch = no wickedd = no bond0.

The Pull request that fixes this issue is here: #27 
 - [x] tested
 - [x] reviewed by SUSE
 - [x] reviewed by SAP

**Comment by vuntz** [comment]
The belief is that https://github.com/crowbar/crowbar-core/pull/1161 is what fixes this.

**Comment by matelakat** [comment]
Needs to be backported and reviewed, and then merged

**Comment by matelakat** [comment]
There is already a PR addressing this issue: #27

**Comment by matelakat** [comment]
I was unable to reproduce the issue with my 6-node setup, so triggered another run with Jenkins, which will do a 10 node installation.

**Comment by matelakat** [comment]
https://ci.suse.de/view/Cloud/view/Cloud6/job/cloud-mkcloud6-job-cd-ha-x86_64/12/
https://ci.suse.de/job/openstack-mkcloud/67407/
Fails on couchdb, investigating

**Comment by matelakat** [comment]
A second run revealed the issue. The build failed right after the `rebootcloud` step. Node 6 did not come back, was not reachable:
```
Error: Waiting for 'node d52-54-77-77-01-06 to be online' timed out.
```
After logging in via vnc console (`virsh vncdisplay vm2-node6`), and doing a clean reboot with the `reboot` command, it came back online.
To reproduce the issue, the node was reset via virsh:
```
virsh reset vm2-node6
```
And the node failed to come back online, although this is not a bonded configuration. As another experience, I tried to re-set `d52-54-77-77-01-07` as well (network-2) to see if that gives the same result:
```
virsh reset vm2-node7
```
The results were the same.
Looking at the results, one interface was missing:
```diff
root@d52-54-77-77-01-06:~ # diff -u mate/*/ip-l-sorted-interfaces.txt
--- mate/good/ip-l-sorted-interfaces.txt        2017-06-14 13:36:34.494425585 +0000
+++ mate/problematic/ip-l-sorted-interfaces.txt 2017-06-14 13:36:48.838420422 +0000
@@ -2,7 +2,6 @@
  br-int: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default
  br-public: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 8900 qdisc noqueue state UNKNOWN mode DEFAULT group default
  br-tunnel: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default
- eth0.200@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 8900 qdisc noqueue state UP mode DEFAULT group default
  eth0.400@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 8900 qdisc noqueue state UP mode DEFAULT group default
  eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 8900 qdisc pfifo_fast master ovs-system state UP mode DEFAULT group default qlen 1000
  lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default

```

**Comment by matelakat** [comment]
To verify that the patch worked, use the following commands:
```
root@crowbar:~ # knife node show d52-54-77-77-01-07.vm2.cloud.suse.de -l | grep -e "ovs_datapath_ids" -A 4
  ovs_datapath_ids:
    br-fixed:   a18b0b3037a93b6f
    br-public:  3eca03a13dc06abf
pacemaker:
  apache2:
root@crowbar:~ # ssh d52-54-77-77-01-07.vm2.cloud.suse.de ovs-vsctl get Bridge br-fixed other-config:datapath-id
"a18b0b3037a93b6f"
root@crowbar:~ # ssh d52-54-77-77-01-07.vm2.cloud.suse.de ovs-vsctl get Bridge br-public other-config:datapath-id
"3eca03a13dc06abf"
```
